djangocms_rest_api/views/base.py

```diff
@@ -1,17 +1,38 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, print_function, unicode_literals
 
+
+from rest_framework.parsers import JSONParser, FormParser, MultiPartParser
+
 from cms.models import Page, Placeholder, CMSPlugin
 from django.contrib.sites.shortcuts import get_current_site
+from django.utils.translation import ugettext as _
+from menus.menu_pool import menu_pool
 from rest_framework import viewsets
+from rest_framework import mixins
+from rest_framework.exceptions import PermissionDenied
 
 from djangocms_rest_api.serializers import (
-    PageSerializer, PlaceHolderSerializer, BasePluginSerializer, get_serializer, get_serializer_class
+    PageSerializer, PlaceHolderSerializer, BasePluginSerializer, get_serializer_class, MenuSerializer
 )
-from djangocms_rest_api.views.utils import QuerysetMixin
+from djangocms_rest_api.views.utils import check_if_page_is_visible
+
+
+class MenuViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    serializer_class = MenuSerializer
+
+    def get_queryset(self):
+        site = get_current_site(self.request)
+        menu_renderer = menu_pool.get_renderer(self.request)
+
+        # next needs to be updated, leave as working for now
+        nodes = menu_renderer._build_nodes(site_id=site.id)
+        nodes = filter(lambda n: not n.parent_id, nodes)
```

Comment:
use pk `nodes = menu_renderer._build_nodes(site_id=site.pk)`
use list comprehension `nodes = [node for node in nodes if not node.parent_id]`